File: 461.hamming-distance.py
#
# @lc app=leetcode id=461 lang=python3
#
# [461] Hamming Distance
#

# @lc code=start
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        z =  x ^ y
        count = 0
        while  z:
            count +=  z % 2
            z =  z // 2
        return count
        
# @lc code=end

File: golang/461.hamming-distance.go
/*
 * @lc app=leetcode.cn id=461 lang=golang
 *
 * [461] Hamming Distance
 */
package main

// @lc code=start
func hammingDistance(x int, y int) int {
	z := x ^ y
	ret := 0
	for z > 0 {
		z &= (z - 1)
		ret++
	}
	return ret
}

// @lc code=end
